scripts/checkpatch.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+DIR="${BASH_SOURCE%/*}"
+if [[ ! -d "$DIR" ]]; then DIR="$PWD"; fi
```

Comment:
How can `DIR` not be a directory? I can see the special case of `checkpatch.sh` being at the root of the filesystem and invoked as `/checkpatch.sh`, in which case `DIR` would be empty. But we don't care about this, since we are in `scripts/`. Any other weird case maybe?